cmd/metal-api/internal/datastore/machine.go

```diff
@@ -443,25 +445,69 @@ func (rs *RethinkStore) FindWaitingMachine(partitionid, sizeid string) (*metal.M
 			"value": string(metal.AvailableState),
 		},
 		"waiting": true,
-	}).Sample(1)
+	})
 
-	var available metal.Machines
-	err := rs.searchEntities(&q, &available)
+	var candidates metal.Machines
+	err := rs.searchEntities(&q, &candidates)
 	if err != nil {
 		return nil, err
 	}
 
-	if len(available) < 1 {
+	var available metal.Machines
+
+	for _, m := range candidates {
+		liveliness, err := rs.EvaluateMachineLiveliness(m)
+		if err != nil {
+			rs.SugaredLogger.Errorw("cannot detect machine liveliness", "machine", m, "error", err)
+			// fall through, so the rest of the machines is getting evaluated
+			continue
+		}
+		switch liveliness {
+		case metal.MachineLivelinessAlive:
+			available = append(available, m)
+		}
+	}
+
+	if available == nil || len(available) < 1 {
 		return nil, fmt.Errorf("no machine available")
 	}
 
-	// we actually return the machine from the machine table, not from the wait table
-	// otherwise we will get in trouble with update operations on the machine table because
-	// we have mixed timestamps with the entity from the wait table...
-	m, err := rs.FindMachineByID(available[0].ID)
```

Comment:
Ha, an old relic from the wait table algorithms. I guess we can kick out this additional call now.